### PR TITLE
Feature/accordion

### PR DESCRIPTION
**Pull request by lbebber**
#### What is the purpose of this pull request?
Adds `submenu.accordion` block.

You can test it on https://lbebber4--storecomponents.myvtex.com/.

Usage is as follows:
```json
  "menu-item": {
    "props": {
        ...
    },
    "blocks": [
      "vtex.menu@2.x:submenu.accordion#name"
    ]
  },
  "vtex.menu@2.x:submenu.accordion#name": {
    "children": [
      "vtex.menu@2.x:menu#menu-name"
    ]
  },  
```
#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/typings/vtex.render-runtime.d.ts [hunk 1]
```diff
@@ -18,11 +18,19 @@ declare module 'vtex.render-runtime' {
...
+    props: BlockProps
+  }
+
   export const ExtensionPoint: ComponentType<ExtensionPointProps>
+  export const useChildBlock: <BlockProps>({id}: {id: string}) => ChildBlock<BlockProps> | null
   export const Helmet: ReactElement
   export const Link: ReactType
   export const NoSSR: ReactElement
-  export const RenderContextConsumer: ReactElement
+  export const RenderContextConsumer: any
```
Comment (lbebber) [inline react/typings/vtex.render-runtime.d.ts]: these should be temporary—indeed, they should not be defined here, but imported from render-runtime

File: react/SubmenuAccordion.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+interface Props {
+  isOpen: boolean
+}
+
+const SubmenuAccordion: StorefrontFunctionComponent<Props> = ({
+  isOpen,
+  children,
+}) => (
+  <Collapsible open={isOpen}>
+    <div className="pl4 flex">
```
Comment (kevinch) [inline react/SubmenuAccordion.tsx]: Useful `div`?

File: react/SubmenuAccordion.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      <section
+        className="w-100 flex"
+        style={{
+          WebkitOverflowScrolling: 'touch',
+          maxHeight: 400,
+          overflowY: 'scroll',
+        }}>
+        <div>
```
Comment (klzns) [inline react/SubmenuAccordion.tsx]: Necessary?

File: react/components/Collapsible.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  private container: RefObject<HTMLDivElement>
+
+  public state = {
+    height: 0,
+  }
+
+  constructor(props: Props) {
+    super(props)
+    this.container = React.createRef()
```
Comment (klzns) [inline react/components/Collapsible.tsx]: move to field property

File: react/SubmenuAccordion.tsx (hunk 2 above)
Comment (lbebber) [reply react/SubmenuAccordion.tsx]: probably has become obsolete, good catch

**Pull request merged**